GetServices/handler.ts

```diff
@@ -83,7 +83,8 @@ export function GetServicesHandler(
                 // keep only the latest version
                 const isNewer =
                   !prev.has(curr.serviceId) ||
-                  curr.version > prev.get(curr.serviceId);
+                  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                  curr.version > prev.get(curr.serviceId)!;
```

Comment:
This is unsafe. The undefined or null value is ignored as if the `strict` option is still disabled.

Reply:
By construction, `prev.get(curr.serviceId)` will never be undefined for the following reasons:
- we've confirmed its existence in the initial part of the predicate, ensuring the `get` function returns a value
- `prev` is a map of NonNegativeInteger type values, which cannot be null

If this explanation suffices, I'll proceed to add a comment.

Reply:
Depends on the previous row, if for a mistake `!prev.has(curr.serviceId)` become `prev.has(curr.serviceId)` the type checker doesn't show any error and the `undefined` result is possible. One of the main goals that TypeScript wants to achieve is to prevent errors.

What if we use something like this?
```typescript
pipe(
  prev.has(curr.serviceId),
  O.fromPredicate(identity),
  O.chainNullableK(() => prev.get(curr.serviceId)),
  O.fold(
    () => true,
    prevVersion => curr.version > prevVersion
  ),
  isNewer =>
    isNewer ? prev.set(curr.serviceId, curr.version) : prev
)
```

Reply:
> Depends on the previous row, if for a mistake !prev.has(curr.serviceId) become prev.has(curr.serviceId) the type checker doesn't show any error and the undefined result is possible.

In this scenario, a test would fail, and I didn't want to change the code.

Anyway, I've updated it, thanks! [[#IOPID-1450] refactor based on comments, part 2](https://github.com/pagopa/io-functions-admin/pull/231/commits/d70ccaa3c845511168ae625d3497e20d15be64ad)